pkg/checkers/tcpping/tcpping.go

```diff
@@ -0,0 +1,167 @@
+package tcpping
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/Azure/kdebug/pkg/base"
+
+	"github.com/fatih/color"
+	"github.com/shirou/gopsutil/process"
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const KubernetesServiceHost = "KUBERNETES_SERVICE_HOST"
+const timeOut = 1000 * time.Millisecond
+
+const (
+	GoogleTarget = "google.com:443"
+)
+
+type pingEndpoint struct {
+	ServerAddress string
+	Name          string
+	NameSpace     string
+}
+
+func (t *TCPChecker) ping(serverAddr string) error {
+	conn, err := t.dialer.Dial("tcp", serverAddr)
+	if err != nil {
+		return err
+	}
+	conn.Close()
```

Comment:
Consider using [TCPConn.SetLinger(0)](https://pkg.go.dev/net#TCPConn.SetLinger) to close the connection faster.

Reply:
fixed